src/prommis/nanofiltration/diafiltration_two_salt.py

```diff
@@ -0,0 +1,1097 @@
+""",
+        ),
+    )
+    CONFIG.declare(
+        "property_package_args",
+        ConfigBlock(
+            implicit=True,
+            description="Arguments to use for constructing property packages",
+            doc="""A ConfigBlock with arguments to be passed to a property block(s)
+and used when constructing these,
+**default** - None.
+**Valid values:** {see property package for documentation}
+""",
+        ),
+    )
+    CONFIG.declare(
+        "NFEx",
+        ConfigValue(
+            doc="Number of discretization points in the x-direction",
+        ),
+    )
+    CONFIG.declare(
+        "NFEz",
+        ConfigValue(
+            doc="Number of discretization points in the z-direction",
+        ),
+    )
+
+    def build(self):
+        """
+        Build method for the two salt diafiltration unit model
+        """
+        super().build()
+
+        # TODO: generalize to any 2 cations and 1 anion
+
+        self.add_mutable_parameters()
+        self.add_variables()
+        self.add_constraints()
+        self.discretize_model()
+        self.add_scaling_factors()
+
+    def add_mutable_parameters(self):
+        """
+        Adds parameters for the two salt diafiltration unit model.
+
+        Assigns default values that can be changed by the user during implementation.
+        """
+        self.membrane_thickness = Param(
+            initialize=1e-7,
+            mutable=True,
+            units=units.m,
+            doc="Thickness of membrane (z-direction)",
+        )
+        self.membrane_width = Param(
+            initialize=1,
+            mutable=True,
+            units=units.m,
+            doc="Width of the membrane (x-direction)",
+        )
+        self.membrane_length = Param(
+            initialize=100,
+            mutable=True,
+            units=units.m,
+            doc="Length of the membrane, wound radially",
+        )
+        self.applied_pressure = Param(
+            initialize=10,
+            mutable=True,
+            units=units.bar,
+            doc="Pressure applied to membrane",
+        )
+        self.membrane_permeability = Param(
+            initialize=0.01,
+            mutable=True,
+            units=units.m / units.h / units.bar,
+            doc="Hydraulic permeability coefficient",
+        )
+        self.feed_flow_volume = Param(
+            initialize=100,
+            mutable=True,
+            units=units.m**3 / units.h,
+            doc="Volumetric flow rate of the feed",
+        )
+        self.feed_conc_mass_lithium = Param(
+            initialize=1.7,
+            mutable=True,
+            units=units.kg / units.m**3,
+            doc="Mass concentration of lithium in the feed",
+        )
+        self.feed_conc_mass_cobalt = Param(
+            initialize=17,
+            mutable=True,
+            units=units.kg / units.m**3,
+            doc="Mass concentration of cobalt in the feed",
+        )
```

Comment:
These also need to be `Vars`, and they need aliases on a port (using `References`) that are compatible with the provided property package.